website/views.py

```diff
@@ -4515,3 +4518,42 @@ def submit_pr(request):
         return render(request, "submit_pr.html")
 
     return render(request, "submit_pr.html")
+
+
+# Global variable to store the vector store
+vector_store = None
+
+
+@api_view(["POST"])
+def question_answer_view(request):
+    question = request.data.get("question", "")
+
+    # apply validation for question , there should be a question and the question should be a string and also the length of the question should be greater than 0
+    if not question or not isinstance(question, str) or len(question) == 0:
+        return Response({"error": "Invalid question"}, status=status.HTTP_400_BAD_REQUEST)
+
+    global vector_store
+    if vector_store is None:
+        db_path = os.path.join(os.path.dirname(__file__), "faiss_index")
+        print(f"Loading FAISS index from: {db_path}")
+        vector_store = load_vector_store(db_path)
+
+    # Handle the "exit" command
+    if question.lower() == "exit":
+        # if buffer is present in the session then delete it
+        if "buffer" in request.session:
+            del request.session["buffer"]
+        return Response({"answer": "Conversation memory cleared."}, status=status.HTTP_200_OK)
+
+    # Initialize session state if not already initialized
+    if "buffer" not in request.session:
+        crc, memory = conversation_chain(vector_store)
+    else:
+        crc, memory = conversation_chain(vector_store)
+        memory.buffer = request.session["buffer"]
```

Comment:
Could you check affirmative conditions first?
```suggestion
    if "buffer" in request.session:
        crc, memory = conversation_chain(vector_store)
        memory.buffer = request.session["buffer"]
    else:
        # Initialize session state if not already initialized.
        crc, memory = conversation_chain(vector_store)
```
```